packages/valory/skills/keep3r_job_abci/models.py

```diff
@@ -133,14 +133,10 @@ def _get_participant_to_swap_pref(self, kwargs: Dict) -> Dict[str, SwapPref]:
                 swap_pref in allowed_swap_prefs,
                 f"Invalid swap pref for participant {participant}!",
             )
-            enforce(
-                participant in all_participants,
-                f"Participant {participant} not in all_participants!",
-            )
             participant_to_swap_pref_dict[participant] = self.SwapPref(swap_pref)
 
         enforce(
-            len(participant_to_swap_pref_dict) == len(all_participants),
+            len(participant_to_swap_pref_dict) > len(all_participants),
```

Comment:
Not sure what is happening here, but shouldn't this be a <? Can we have more prefs than participants?

Reply:
Idea is that we need to specifiy preferences for all (historical) participants + service owner. This means that the length of preferences MUST always be larger than the number of participants.

Reply:
Line 131 should be:

```python
for participant, swap_pref in participant_to_swap_pref.items():
```

instead of:
https://github.com/valory-xyz/agent-academy-2/blob/324c58a6e9b3196f0d840ff81227412609775a2b/packages/valory/skills/keep3r_job_abci/models.py#L131